src/commands/createVirtualMachine/ConfirmPassphraseStep.ts

```diff
@@ -0,0 +1,37 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { AzureWizardPromptStep } from "vscode-azureextensionui";
+import { ext } from "../../extensionVariables";
+import { localize } from "../../localize";
+import { nonNullProp } from "../../utils/nonNull";
+import { IVirtualMachineWizardContext } from "./IVirtualMachineWizardContext";
+
+let mismatchErrorThrown: boolean = false;
+export class ConfirmPassphraseStep extends AzureWizardPromptStep<IVirtualMachineWizardContext> {
+    public async prompt(wizardContext: IVirtualMachineWizardContext): Promise<void> {
+        const prompt: string = localize('confirmPassphrase', 'Confirm your passphrase');
+        await ext.ui.showInputBox({
+            prompt,
+            password: true,
+            validateInput: async (value: string | undefined): Promise<string | undefined> => await this.validatePassphrase(wizardContext, value)
+        });
+    }
+
+    public shouldPrompt(wizardContext: IVirtualMachineWizardContext): boolean {
+        return !!wizardContext.passphrase;
+    }
+
+    private async validatePassphrase(wizardContext: IVirtualMachineWizardContext, passphrase: string | undefined): Promise<string | undefined> {
+        if (passphrase === wizardContext.passphrase) {
```

Comment:
I added this extra logic because I _HATE_ when the password confirming starts yelling at me before I could have feasibly typed in my password yet.

This will make sure that the error won't be thrown until the user has at least typed enough characters to have typed the password to begin wtih.

Reply:
I see your frustration, but this means users can type a non-matching password (less than 5 characters). I don't know how to prevent that and fix your frustration.

What if you just throw an error saying "You're getting closer!" until they reach the right length (I'm kidding)